app/master/build.py

```diff
@@ -326,6 +340,10 @@ def cancel(self):
         self._state_machine.trigger(BuildEvent.CANCEL)
 
     def _on_enter_canceled_state(self, event):
+        # Set the kill_event to kill the subprocesses for the build
+        self._logger.notice('Set the kill_event for subprocesses for build {}.', self._build_id)
+        self.project_type.kill_subprocesses()
+
```

Comment:
Previously we were not killing the subprocesses for the build namely atomizer. We need to kill the subprocesses on cancel.
I am not 100% sure about draining of `unstarted_subjobs`. Do you want to say that as subprocesses are killed, it does not matter? My opinion is its a good idea to clean in any case.